lib/net/Makefile

```diff
@@ -5,20 +5,33 @@ NETIFFILES := $(filter-out $(LWIPDIR)/netif/slipif.c,$(NETIFFILES))
 LWIPSRCS := $(COREFILES) \
             $(CORE4FILES) \
             $(CORE6FILES) \
-            $(NETIFFILES) \
-            $(APIFILES)
+            $(APIFILES) \
+            $(NETIFFILES)
 
 # Include driver sources
 DRIVERSRCS := $(NXDK_DIR)/lib/net/pktdrv/pktdrv.c \
               $(NXDK_DIR)/lib/net/nforceif/src/driver.c \
               $(NXDK_DIR)/lib/net/nforceif/src/sys_arch.c
 
-SRCS += $(LWIPSRCS) $(DRIVERSRCS)
+NET_SRCS := \
```

Comment:
*(This is a generic comment about PR description, but made on code so we can resolve it more easily)*

> - The include `CFLAGS` for net are in the net-Makefile, I could have moved them into `nxdk/Makefile`(following the other libs examples(hal, pdclib, winapi, etc.). But that feels wrong - so I wonder, why are the other libs included there and not in their own Makefile(i.e. like libpng does)?

The network stuff are optional includes and only necessary if the user actually needs network access.
So adding them here is better, because this Makefile is conditionally included.

The others are part of making a C/Windows toolchain, so they are always mandatory (although we might eventually want a `-freestanding` option for something like my bin-wrappers).